presto-hive/src/test/java/com/facebook/presto/hive/TestHiveMaterializedViewLogicalPlanner.java

```diff
@@ -1471,6 +1474,294 @@ public void testMaterializedViewForJoin()
+                    "SELECT suppkey, SUM(quantity) AS qty \n " +
+                    "FROM %s \n " +
+                    "GROUP BY suppkey, ds) \n " +
+                    "SELECT n.suppkey, n.name, r.returned_qty, s.qty \n " +
+                    "FROM long_name_supp AS n \n " +
+                    "LEFT JOIN supp_returns AS r ON n.suppkey = r.suppkey \n " +
+                    "LEFT JOIN supp_sum AS s ON n.suppkey = s.suppkey \n " +
+                    "ORDER BY suppkey, name",
+                    supplierTable, lineItemTable, lineItemTable);
+
+            MaterializedResult optimizedQueryResult = computeActual(queryOptimizationWithMaterializedView, baseQuery);
+            MaterializedResult baseQueryResult = computeActual(baseQuery);
+            assertEquals(baseQueryResult, optimizedQueryResult);
+
+            PlanMatchPattern expectedPattern = anyTree(
+                    node(JoinNode.class,
+                            node(JoinNode.class,
+                                    exchange(
+                                            anyTree(
+                                                    constrainedTableScan(supplierTable,
+                                                            ImmutableMap.of("ds", multipleValues(createVarcharType(10), utf8Slices("2021-07-12"))),
+                                                            ImmutableMap.of())),
+                                            anyTree(
+                                                    constrainedTableScan(suppliersView,
+                                                            ImmutableMap.of("ds", multipleValues(createVarcharType(10), utf8Slices("2021-07-11"))),
+                                                            ImmutableMap.of("suppkey_37", "suppkey", "name_38", "name")))),
+                                    anyTree(
+                                            constrainedTableScan(lineItemTable,
+                                                    ImmutableMap.of("ds", multipleValues(createVarcharType(10), utf8Slices("2021-07-11", "2021-07-12"))),
+                                                    ImmutableMap.of("ds_73", "ds", "suppkey_71", "suppkey")))),
+                            exchange(
+                                    anyTree(
+                                            exchange(
+                                                    anyTree(
+                                                            constrainedTableScan(lineItemTable,
+                                                                    ImmutableMap.of("ds", multipleValues(createVarcharType(10), utf8Slices("2021-07-11"))),
+                                                                    ImmutableMap.of("ds_108", "ds", "suppkey_94", "suppkey", "quantity_96", "quantity"))),
+                                                    anyTree(
+                                                            constrainedTableScan(lineItemView1,
```

Comment:
I wonder why do we need to specify all dates in this plan comparision, we are effectively reading everything present in the view, so this comparision could be simpler by not providing any ds constraints, right? This applies to other comparision as well.

Reply:
@Sullivan-Patrick I see this as resolved, but see similar changes, am I missing something?